Unable to install on Raspberry Pi

OS: Raspbian 
BitcoinD: 0.16.0
LND 0.4.2
Node.js version: 10.4.1
npm version: 6.1.0
Yarn: 1.7.0


yarn
yarn install v1.7.0
[1/5] Validating package.json...
[2/5] Resolving packages...
success Already up-to-date.
$ concurrently "npm run flow-typed" "npm run build-dll" "electron-builder install-app-deps" "node node_modules/fbjs-scripts/node/check-dev-engines.js package.json"
[1] 
[1] > zap-desktop@0.1.1 build-dll /home/pi/source/zap-desktop
[1] > cross-env NODE_ENV=development node --trace-warnings -r babel-register ./node_modules/webpack/bin/webpack --config webpack.config.renderer.dev.dll.js --colors
[1] 
[0] 
[0] > zap-desktop@0.1.1 flow-typed /home/pi/source/zap-desktop
[0] > rimraf flow-typed/npm && flow-typed install --overwrite || true
[0] 
[3] node node_modules/fbjs-scripts/node/check-dev-engines.js package.json exited with code 0
[2] electron-builder install-app-deps
[2] 
[2] Install app deps
[2] 
[2] Options:
[2]   --version   Show version number                                      [boolean]
[2]   --help      Show help                                                [boolean]
[2]   --platform  The target platform
[2]                         [choices: "linux", "darwin", "win32"] [default: "linux"]
[2]   --arch      The target arch
[2]              [choices: "ia32", "x64", "armv7l", "arm64", "all"] [default: "arm"]
[2] 
[2] Invalid values:
[2]   Argument: arch, Given: "arm", Choices: "ia32", "x64", "armv7l", "arm64", "all"
[2] electron-builder install-app-deps exited with code 1
[0] • Found 101 dependencies in package.json to install libdefs for. Searching...
[0] • flow-typed cache not found, fetching from GitHub...
[1] Hash: 8fa2e9246a73ce6dba61
[1] Version: webpack 3.7.1
[1] Time: 27341ms
[1]               Asset     Size  Chunks                    Chunk Names
[1] renderer.dev.dll.js  4.58 MB       0  [emitted]  [big]  renderer
[1] [./node_modules/moment/locale recursive ^\.\/.*$] ./node_modules/moment/locale ^\.\/.*$ 2.98 kB {0} [optional] [built]
[1] [./node_modules/webpack/buildin/harmony-module.js] (webpack)/buildin/harmony-module.js 596 bytes {0} [built]
[1] [./node_modules/webpack/buildin/module.js] (webpack)/buildin/module.js 517 bytes {0} [built]
[1]    [0] dll renderer 12 bytes {0} [built]
[1]     + 852 hidden modules
[1] npm run build-dll exited with code 0
[0] • Installing 18 libDefs...
[0]   • axios_v0.16.x.js
[0]     └> ./flow-typed/npm/axios_v0.16.x.js
[0] axios
[0] /home/pi/source/zap-desktop/flow-typed/npm
[0]   • enzyme_v2.3.x.js
[0]     └> ./flow-typed/npm/enzyme_v2.3.x.js
[0] enzyme
[0] /home/pi/source/zap-desktop/flow-typed/npm
[0]   • flow-bin_v0.x.x.js
[0]     └> ./flow-typed/npm/flow-bin_v0.x.x.js
[0] flow-bin
[0] /home/pi/source/zap-desktop/flow-typed/npm
[0]   • express_v4.x.x.js
[0]     └> ./flow-typed/npm/express_v4.x.x.js
[0] express
[0] /home/pi/source/zap-desktop/flow-typed/npm
[0]   • minimist_v1.x.x.js
[0]     └> ./flow-typed/npm/minimist_v1.x.x.js
[0] minimist
[0] /home/pi/source/zap-desktop/flow-typed/npm
[0]   • copy-to-clipboard_v3.x.x.js
[0]     └> ./flow-typed/npm/copy-to-clipboard_v3.x.x.js
[0] copy-to-clipboard
[0] /home/pi/source/zap-desktop/flow-typed/npm
[0]   • redux_v3.x.x.js
[0]     └> ./flow-typed/npm/redux_v3.x.x.js
[0] redux
[0] /home/pi/source/zap-desktop/flow-typed/npm
[0]   • chalk_v2.x.x.js
[0]     └> ./flow-typed/npm/chalk_v2.x.x.js
[0] chalk
[0] /home/pi/source/zap-desktop/flow-typed/npm
[0]   • history_v4.x.x.js
[0]     └> ./flow-typed/npm/history_v4.x.x.js
[0] history
[0] /home/pi/source/zap-desktop/flow-typed/npm
[0]   • jest_v20.x.x.js
[0]     └> ./flow-typed/npm/jest_v20.x.x.js
[0] jest
[0] /home/pi/source/zap-desktop/flow-typed/npm
[0]   • rimraf_v2.x.x.js
[0]     └> ./flow-typed/npm/rimraf_v2.x.x.js
[0] rimraf
[0] /home/pi/source/zap-desktop/flow-typed/npm
[0]   • lodash_v4.x.x.js
[0]     └> ./flow-typed/npm/lodash_v4.x.x.js
[0] lodash
[0] /home/pi/source/zap-desktop/flow-typed/npm
[0]   • react-redux_v5.x.x.js
[0]     └> ./flow-typed/npm/react-redux_v5.x.x.js
[0] react-redux
[0] /home/pi/source/zap-desktop/flow-typed/npm
[0]   • reselect_v3.x.x.js
[0]     └> ./flow-typed/npm/reselect_v3.x.x.js
[0] reselect
[0] /home/pi/source/zap-desktop/flow-typed/npm
[0]   • react-router_v4.x.x.js
[0]     └> ./flow-typed/npm/react-router_v4.x.x.js
[0] react-router
[0] /home/pi/source/zap-desktop/flow-typed/npm
[0]   • prop-types_v15.x.x.js
[0]     └> ./flow-typed/npm/prop-types_v15.x.x.js
[0] prop-types
[0] /home/pi/source/zap-desktop/flow-typed/npm
[0]   • react-router-dom_v4.x.x.js
[0]     └> ./flow-typed/npm/react-router-dom_v4.x.x.js
[0] react-router-dom
[0] /home/pi/source/zap-desktop/flow-typed/npm
[0]   • react-addons-test-utils_v15.x.x.js
[0]     └> ./flow-typed/npm/react-addons-test-utils_v15.x.x.js
[0] react-addons-test-utils
[0] /home/pi/source/zap-desktop/flow-typed/npm
[0] • Generating stubs for untyped dependencies...
[0]   • satoshi-bitcoin@^1.0.4
[0]     └> flow-typed/npm/satoshi-bitcoin_vx.x.x.js
[0]   • xtend@^4.0.1
[0]     └> flow-typed/npm/xtend_vx.x.x.js
[0]   • electron-store@^1.3.0
[0]     └> flow-typed/npm/electron-store_vx.x.x.js
[0]   • electron-debug@^1.2.0
[0]     └> flow-typed/npm/electron-debug_vx.x.x.js
[0]   • source-map-support@^0.4.15
[0]     └> flow-typed/npm/source-map-support_vx.x.x.js
[0]   • browserslist@^2.11.0
[0]     └> flow-typed/npm/browserslist_vx.x.x.js
[0]   • babel-plugin-dev-expression@^0.2.1
[0]     └> flow-typed/npm/babel-plugin-dev-expression_vx.x.x.js
[0]   • eslint-import-resolver-webpack@^0.8.4
[0]     └> flow-typed/npm/eslint-import-resolver-webpack_vx.x.x.js
[0]   • file-loader@^0.11.1
[0]     └> flow-typed/npm/file-loader_vx.x.x.js
[0]   • eslint-formatter-pretty@^1.3.0
[0]     └> flow-typed/npm/eslint-formatter-pretty_vx.x.x.js
[0]   • stylelint-config-standard@^18.2.0
[0]     └> flow-typed/npm/stylelint-config-standard_vx.x.x.js
[0]   • url-loader@^0.5.8
[0]     └> flow-typed/npm/url-loader_vx.x.x.js
[0]   • d3-force@^1.1.0
[0]     └> flow-typed/npm/d3-force_vx.x.x.js
[0]   • d3-zoom@^1.7.1
[0]     └> flow-typed/npm/d3-zoom_vx.x.x.js
[0]   • qrcode.react@0.8.0
[0]     └> flow-typed/npm/qrcode.react_vx.x.x.js
[0]   • react-inlinesvg@^0.6.2
[0]     └> flow-typed/npm/react-inlinesvg_vx.x.x.js
[0]   • redux-thunk@^2.2.0
[0]     └> flow-typed/npm/redux-thunk_vx.x.x.js
[0]   • react-router-redux@^5.0.0-alpha.6
[0]     └> flow-typed/npm/react-router-redux_vx.x.x.js
[0]   • react-moment@^0.6.0
[0]     └> flow-typed/npm/react-moment_vx.x.x.js
[0]   • babel-plugin-add-module-exports@^0.2.1
[0]     └> flow-typed/npm/babel-plugin-add-module-exports_vx.x.x.js
[0]   • redux-electron-ipc@^1.1.10
[0]     └> flow-typed/npm/redux-electron-ipc_vx.x.x.js
[0]   • babel-jest@^20.0.3
[0]     └> flow-typed/npm/babel-jest_vx.x.x.js
[0]   • babel-preset-env@^1.5.1
[0]     └> flow-typed/npm/babel-preset-env_vx.x.x.js
[0]   • babel-plugin-dynamic-import-webpack@^1.0.1
[0]     └> flow-typed/npm/babel-plugin-dynamic-import-webpack_vx.x.x.js
[0]   • babel-preset-react@^6.24.1
[0]     └> flow-typed/npm/babel-preset-react_vx.x.x.js
[0]   • font-awesome@^4.7.0
[0]     └> flow-typed/npm/font-awesome_vx.x.x.js
[0]   • babel-preset-react-optimize@^1.0.1
[0]     └> flow-typed/npm/babel-preset-react-optimize_vx.x.x.js
[0]   • babel-preset-stage-0@^6.24.1
[0]     └> flow-typed/npm/babel-preset-stage-0_vx.x.x.js
[0]   • babel-register@^6.24.1
[0]     └> flow-typed/npm/babel-register_vx.x.x.js
[0]   • babili-webpack-plugin@^0.1.2
[0]     └> flow-typed/npm/babili-webpack-plugin_vx.x.x.js
[0]   • css-loader@^0.28.3
[0]     └> flow-typed/npm/css-loader_vx.x.x.js
[0]   • babel-plugin-transform-class-properties@^6.24.1
[0]     └> flow-typed/npm/babel-plugin-transform-class-properties_vx.x.x.js
[0]   • electron-devtools-installer@^2.2.1
[0]     └> flow-typed/npm/electron-devtools-installer_vx.x.x.js
[0]   • eslint-config-airbnb@^16.1.0
[0]     └> flow-typed/npm/eslint-config-airbnb_vx.x.x.js
[0]   • html-webpack-plugin@^2.29.0
[0]     └> flow-typed/npm/html-webpack-plugin_vx.x.x.js
[0]   • enzyme-to-json@^1.5.1
[0]     └> flow-typed/npm/enzyme-to-json_vx.x.x.js
[0]   • ps-node@^0.1.6
[0]     └> flow-typed/npm/ps-node_vx.x.x.js
[0]   • redux-logger@^3.0.6
[0]     └> flow-typed/npm/redux-logger_vx.x.x.js
[0]   • sass-loader@^6.0.6
[0]     └> flow-typed/npm/sass-loader_vx.x.x.js
[0]   • spectron@^3.8.0
[0]     └> flow-typed/npm/spectron_vx.x.x.js
[0]   • style-loader@^0.18.1
[0]     └> flow-typed/npm/style-loader_vx.x.x.js
[0]   • babel-preset-react-hmre@^1.1.1
[0]     └> flow-typed/npm/babel-preset-react-hmre_vx.x.x.js
[0]   • webpack-merge@^4.1.0
[0]     └> flow-typed/npm/webpack-merge_vx.x.x.js
[0]   • electron@1.8.4
[0]     └> flow-typed/npm/electron_vx.x.x.js
[0]   • react-modal@^2.2.2
[0]     └> flow-typed/npm/react-modal_vx.x.x.js
[0]   • react-hot-loader@3.0.0-beta.6
[0]     └> flow-typed/npm/react-hot-loader_vx.x.x.js
[0]   • moment-timezone@^0.5.13
[0]     └> flow-typed/npm/moment-timezone_vx.x.x.js
[0]   • babel-loader@^7.1.0
[0]     └> flow-typed/npm/babel-loader_vx.x.x.js
[0]   • babel-eslint@^8.2.1
[0]     └> flow-typed/npm/babel-eslint_vx.x.x.js
[0]   • cross-env@^5.0.0
[0]     └> flow-typed/npm/cross-env_vx.x.x.js
[0]   • electron-builder@^20.15.1
[0]     └> flow-typed/npm/electron-builder_vx.x.x.js
[0]   • concurrently@^3.5.0
[0]     └> flow-typed/npm/concurrently_vx.x.x.js
[0]   • eslint-plugin-import@^2.8.0
[0]     └> flow-typed/npm/eslint-plugin-import_vx.x.x.js
[0]   • eslint-plugin-compat@^2.2.0
[0]     └> flow-typed/npm/eslint-plugin-compat_vx.x.x.js
[0]   • babel-plugin-transform-es2015-classes@^6.24.1
[0]     └> flow-typed/npm/babel-plugin-transform-es2015-classes_vx.x.x.js
[0]   • eslint-plugin-promise@^3.6.0
[0]     └> flow-typed/npm/eslint-plugin-promise_vx.x.x.js
[0]   • eslint-plugin-react@^7.6.1
[0]     └> flow-typed/npm/eslint-plugin-react_vx.x.x.js
[0]   • extract-text-webpack-plugin@^3.0.1
[0]     └> flow-typed/npm/extract-text-webpack-plugin_vx.x.x.js
[0]   • cross-spawn@^5.1.0
[0]     └> flow-typed/npm/cross-spawn_vx.x.x.js
[0]   • identity-obj-proxy@^3.0.0
[0]     └> flow-typed/npm/identity-obj-proxy_vx.x.x.js
[0]   • react-test-renderer@^15.6.1
[0]     └> flow-typed/npm/react-test-renderer_vx.x.x.js
[0]   • webpack-bundle-analyzer@^2.8.2
[0]     └> flow-typed/npm/webpack-bundle-analyzer_vx.x.x.js
[0]   • webpack@^3.7.1
[0]     └> flow-typed/npm/webpack_vx.x.x.js
[0]   • fbjs-scripts@^0.8.0
[0]     └> flow-typed/npm/fbjs-scripts_vx.x.x.js
[0]   • d3-selection@^1.2.0
[0]     └> flow-typed/npm/d3-selection_vx.x.x.js
[0]   • devtron@^1.4.0
[0]     └> flow-typed/npm/devtron_vx.x.x.js
[0]   • bitcoinjs-lib@^3.2.0
[0]     └> flow-typed/npm/bitcoinjs-lib_vx.x.x.js
[0]   • eslint@^4.16.0
[0]     └> flow-typed/npm/eslint_vx.x.x.js
[0]   • eslint-plugin-flowtype@^2.42.0
[0]     └> flow-typed/npm/eslint-plugin-flowtype_vx.x.x.js
[0]   • eslint-plugin-jest@^21.7.0
[0]     └> flow-typed/npm/eslint-plugin-jest_vx.x.x.js
[0]   • flow-typed@^2.1.2
[0]     └> flow-typed/npm/flow-typed_vx.x.x.js
[0]   • bitcore-lib@^0.14.0
[0]     └> flow-typed/npm/bitcore-lib_vx.x.x.js
[0]   • webpack-dev-server@^2.5.0
[0]     └> flow-typed/npm/webpack-dev-server_vx.x.x.js
[0]   • stylelint@^9.1.1
[0]     └> flow-typed/npm/stylelint_vx.x.x.js
[0]   • babel-core@^6.24.1
[0]     └> flow-typed/npm/babel-core_vx.x.x.js
[0]   • eslint-plugin-jsx-a11y@6.0.3
[0]     └> flow-typed/npm/eslint-plugin-jsx-a11y_vx.x.x.js
[0]   • flow-runtime@^0.13.0
[0]     └> flow-typed/npm/flow-runtime_vx.x.x.js
[0]   • jsdom@^11.0.0
[0]     └> flow-typed/npm/jsdom_vx.x.x.js
[0]   • sinon@^2.3.5
[0]     └> flow-typed/npm/sinon_vx.x.x.js
[0]   • babel-plugin-flow-runtime@^0.11.1
[0]     └> flow-typed/npm/babel-plugin-flow-runtime_vx.x.x.js
[0]   • node-sass@^4.9.0
[0]     └> flow-typed/npm/node-sass_vx.x.x.js
[0] 
[0] !! No flow@v0.48.0-compatible libdefs found in flow-typed for the above untyped dependencies !!
[0] 
[0] I've generated `any`-typed stubs for these packages, but consider submitting 
[0] libdefs for them to https://github.com/flowtype/flow-typed/
[0] 
[0] npm run flow-typed exited with code 0
error Command failed with exit code 1.

*******************************************************
pi@raspberrypi:~/source/zap-desktop $ npm run install-grpc

> zap-desktop@0.1.1 install-grpc /home/pi/source/zap-desktop
> cd app && npm run install-grpc


> zap-desktop@0.1.1 install-grpc /home/pi/source/zap-desktop/app
> cd node_modules/grpc && git submodule update --init && npm run electron-build -- --target=1.8.4

sh: 1: cd: can't cd to node_modules/grpc
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! zap-desktop@0.1.1 install-grpc: `cd node_modules/grpc && git submodule update --init && npm run electron-build -- --target=1.8.4`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the zap-desktop@0.1.1 install-grpc script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm WARN Local package.json exists, but node_modules missing, did you mean to install?

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/pi/.npm/_logs/2018-06-14T08_10_40_302Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! zap-desktop@0.1.1 install-grpc: `cd app && npm run install-grpc`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the zap-desktop@0.1.1 install-grpc script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/pi/.npm/_logs/2018-06-14T08_10_40_372Z-debug.log

npx electron-builder install-app-deps --arch armv7l

Then there was a problem with grpc.

npm install grpc

Then there was another problem with grpc:
"Unable to CD node_Modules/grpc"  
The directory is clearly there.

Manually:
cd node_Modules/grpc
git submodule update --init && npm run electron-build -- --target=1.8.4

This completed with no errors.


pi@raspberrypi:~/source/zap-desktop $ npm run test

> zap-desktop@0.1.1 test /home/pi/source/zap-desktop
> cross-env NODE_ENV=test BABEL_DISABLE_CACHE=1 node --trace-warnings ./test/runTests.js

internal/modules/cjs/loader.js:596
    throw err;
    ^

Error: Cannot find module '/home/pi/source/zap-desktop/node_modules/jest-cli'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:594:15)
    at Function.Module._load (internal/modules/cjs/loader.js:520:25)
    at Module.require (internal/modules/cjs/loader.js:650:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at getJest (/home/pi/source/zap-desktop/node_modules/jest/node_modules/jest-cli/build/cli/getJest.js:22:12)
    at Object.run (/home/pi/source/zap-desktop/node_modules/jest/node_modules/jest-cli/build/cli/index.js:39:48)
    at Object.<anonymous> (/home/pi/source/zap-desktop/node_modules/jest/node_modules/jest-cli/bin/jest.js:14:25)
    at Module._compile (internal/modules/cjs/loader.js:702:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:713:10)
    at Module.load (internal/modules/cjs/loader.js:612:32)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! zap-desktop@0.1.1 test: `cross-env NODE_ENV=test BABEL_DISABLE_CACHE=1 node --trace-warnings ./test/runTests.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the zap-desktop@0.1.1 test script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/pi/.npm/_logs/2018-06-17T01_45_43_331Z-debug.log
pi@raspberrypi:~/source/zap-desktop $ cd node_modules/
pi@raspberrypi:~/source/zap-desktop/node_modules $ cd jest-cli
pi@raspberrypi:~/source/zap-desktop/node_modules/jest-cli $ ls
node_modules
pi@raspberrypi:~/source/zap-desktop/node_modules/jest-cli $ cd node_modules/
pi@raspberrypi:~/source/zap-desktop/node_modules/jest-cli/node_modules $ ls
pi@raspberrypi:~/source/zap-desktop/node_modules/jest-cli/node_modules $ cd ..
pi@raspberrypi:~/source/zap-desktop/node_modules/jest-cli $ cd..
bash: cd..: command not found
pi@raspberrypi:~/source/zap-desktop/node_modules/jest-cli $ cd ..
pi@raspberrypi:~/source/zap-desktop/node_modules $ cd ..
pi@raspberrypi:~/source/zap-desktop $ npm run dev

> zap-desktop@0.1.1 dev /home/pi/source/zap-desktop
> cross-env START_HOT=1 concurrently "npm run start-renderer-dev"

[0] 
[0] > zap-desktop@0.1.1 start-renderer-dev /home/pi/source/zap-desktop
[0] > cross-env NODE_ENV=development node --trace-warnings -r babel-register ./node_modules/webpack-dev-server/bin/webpack-dev-server --config webpack.config.renderer.dev.js
[0] 
[0] Starting Main Process...
[0] Project is running at http://localhost:1212/
[0] webpack output is served from http://localhost:1212/dist
[0] Content not from webpack is served from /home/pi/source/zap-desktop/dist
[0] 404s will fallback to /index.html
[0] 
[0] > zap-desktop@0.1.1 start-main-dev /home/pi/source/zap-desktop
[0] > cross-env HOT=1 NODE_ENV=development electron -r babel-register ./app/main.dev
[0] 
[0] 
[0] ERROR in ./app/components/Activity/TransactionModal.js
[0] Module not found: Error: Can't resolve 'react-icons/lib/fa' in '/home/pi/source/zap-desktop/app/components/Activity'
[0]  @ ./app/components/Activity/TransactionModal.js 21:10-39
[0]  @ ./app/components/Activity/ActivityModal.js
[0]  @ ./app/routes/app/components/App.js
[0]  @ ./app/routes/app/containers/AppContainer.js
[0]  @ ./app/routes/app/index.js
[0]  @ ./app/routes.js
[0]  @ ./app/containers/Root.js
[0]  @ ./app/index.js
[0]  @ multi (webpack)-dev-server/client?http://localhost:1212 webpack/hot/dev-server react-hot-loader/patch webpack-dev-server/client?http://localhost:1212/ webpack/hot/only-dev-server ./app/index.js
[0] 
[0] ERROR in ./app/components/Activity/InvoiceModal.js
[0] Module not found: Error: Can't resolve 'react-icons/lib/fa' in '/home/pi/source/zap-desktop/app/components/Activity'
[0]  @ ./app/components/Activity/InvoiceModal.js 31:10-39
[0]  @ ./app/components/Activity/ActivityModal.js
[0]  @ ./app/routes/app/components/App.js
[0]  @ ./app/routes/app/containers/AppContainer.js
[0]  @ ./app/routes/app/index.js
[0]  @ ./app/routes.js
[0]  @ ./app/containers/Root.js
[0]  @ ./app/index.js
[0]  @ multi (webpack)-dev-server/client?http://localhost:1212 webpack/hot/dev-server react-hot-loader/patch webpack-dev-server/client?http://localhost:1212/ webpack/hot/only-dev-server ./app/index.js
[0] 
[0] ERROR in ./app/components/Activity/PaymentModal.js
[0] Module not found: Error: Can't resolve 'react-icons/lib/fa' in '/home/pi/source/zap-desktop/app/components/Activity'
[0]  @ ./app/components/Activity/PaymentModal.js 21:10-39
[0]  @ ./app/components/Activity/ActivityModal.js
[0]  @ ./app/routes/app/components/App.js
[0]  @ ./app/routes/app/containers/AppContainer.js
[0]  @ ./app/routes/app/index.js
[0]  @ ./app/routes.js
[0]  @ ./app/containers/Root.js
[0]  @ ./app/index.js
[0]  @ multi (webpack)-dev-server/client?http://localhost:1212 webpack/hot/dev-server react-hot-loader/patch webpack-dev-server/client?http://localhost:1212/ webpack/hot/only-dev-server ./app/index.js
[0] 
[0] ERROR in ./app/components/Contacts/ContactModal.js
[0] Module not found: Error: Can't resolve 'react-icons/lib/fa' in '/home/pi/source/zap-desktop/app/components/Contacts'
[0]  @ ./app/components/Contacts/ContactModal.js 23:10-39
[0]  @ ./app/routes/app/components/App.js
[0]  @ ./app/routes/app/containers/AppContainer.js
[0]  @ ./app/routes/app/index.js
[0]  @ ./app/routes.js
[0]  @ ./app/containers/Root.js
[0]  @ ./app/index.js
[0]  @ multi (webpack)-dev-server/client?http://localhost:1212 webpack/hot/dev-server react-hot-loader/patch webpack-dev-server/client?http://localhost:1212/ webpack/hot/only-dev-server ./app/index.js
[0] 
[0] ERROR in ./app/components/Contacts/Network.js
[0] Module not found: Error: Can't resolve 'react-icons/lib/fa' in '/home/pi/source/zap-desktop/app/components/Contacts'
[0]  @ ./app/components/Contacts/Network.js 25:10-39
[0]  @ ./app/routes/app/components/App.js
[0]  @ ./app/routes/app/containers/AppContainer.js
[0]  @ ./app/routes/app/index.js
[0]  @ ./app/routes.js
[0]  @ ./app/containers/Root.js
[0]  @ ./app/index.js
[0]  @ multi (webpack)-dev-server/client?http://localhost:1212 webpack/hot/dev-server react-hot-loader/patch webpack-dev-server/client?http://localhost:1212/ webpack/hot/only-dev-server ./app/index.js
[0] 
[0] ERROR in ./app/components/Contacts/SubmitChannelForm.js
[0] Module not found: Error: Can't resolve 'react-icons/lib/fa' in '/home/pi/source/zap-desktop/app/components/Contacts'
[0]  @ ./app/components/Contacts/SubmitChannelForm.js 17:10-39
[0]  @ ./app/components/Contacts/ChannelForm.js
[0]  @ ./app/routes/app/components/App.js
[0]  @ ./app/routes/app/containers/AppContainer.js
[0]  @ ./app/routes/app/index.js
[0]  @ ./app/routes.js
[0]  @ ./app/containers/Root.js
[0]  @ ./app/index.js
[0]  @ multi (webpack)-dev-server/client?http://localhost:1212 webpack/hot/dev-server react-hot-loader/patch webpack-dev-server/client?http://localhost:1212/ webpack/hot/only-dev-server ./app/index.js
[0] 
[0] ERROR in ./app/components/Form/Pay.js
[0] Module not found: Error: Can't resolve 'react-icons/lib/fa' in '/home/pi/source/zap-desktop/app/components/Form'
[0]  @ ./app/components/Form/Pay.js 33:10-39
[0]  @ ./app/components/Form/Form.js
[0]  @ ./app/components/Form/index.js
[0]  @ ./app/routes/app/components/App.js
[0]  @ ./app/routes/app/containers/AppContainer.js
[0]  @ ./app/routes/app/index.js
[0]  @ ./app/routes.js
[0]  @ ./app/containers/Root.js
[0]  @ ./app/index.js
[0]  @ multi (webpack)-dev-server/client?http://localhost:1212 webpack/hot/dev-server react-hot-loader/patch webpack-dev-server/client?http://localhost:1212/ webpack/hot/only-dev-server ./app/index.js
[0] 
[0] ERROR in ./app/components/Form/Request.js
[0] Module not found: Error: Can't resolve 'react-icons/lib/fa' in '/home/pi/source/zap-desktop/app/components/Form'
[0]  @ ./app/components/Form/Request.js 23:10-39
[0]  @ ./app/components/Form/Form.js
[0]  @ ./app/components/Form/index.js
[0]  @ ./app/routes/app/components/App.js
[0]  @ ./app/routes/app/containers/AppContainer.js
[0]  @ ./app/routes/app/index.js
[0]  @ ./app/routes.js
[0]  @ ./app/containers/Root.js
[0]  @ ./app/index.js
[0]  @ multi (webpack)-dev-server/client?http://localhost:1212 webpack/hot/dev-server react-hot-loader/patch webpack-dev-server/client?http://localhost:1212/ webpack/hot/only-dev-server ./app/index.js
[0] 
[0] ERROR in ./app/components/Onboarding/FormContainer.js
[0] Module not found: Error: Can't resolve 'react-icons/lib/fa' in '/home/pi/source/zap-desktop/app/components/Onboarding'
[0]  @ ./app/components/Onboarding/FormContainer.js 19:10-39
[0]  @ ./app/components/Onboarding/Onboarding.js
[0]  @ ./app/components/Onboarding/index.js
[0]  @ ./app/containers/Root.js
[0]  @ ./app/index.js
[0]  @ multi (webpack)-dev-server/client?http://localhost:1212 webpack/hot/dev-server react-hot-loader/patch webpack-dev-server/client?http://localhost:1212/ webpack/hot/only-dev-server ./app/index.js
[0] 
[0] ERROR in ./app/components/Onboarding/Signup.js
[0] Module not found: Error: Can't resolve 'react-icons/lib/fa' in '/home/pi/source/zap-desktop/app/components/Onboarding'
[0]  @ ./app/components/Onboarding/Signup.js 15:10-39
[0]  @ ./app/components/Onboarding/Onboarding.js
[0]  @ ./app/components/Onboarding/index.js
[0]  @ ./app/containers/Root.js
[0]  @ ./app/index.js
[0]  @ multi (webpack)-dev-server/client?http://localhost:1212 webpack/hot/dev-server react-hot-loader/patch webpack-dev-server/client?http://localhost:1212/ webpack/hot/only-dev-server ./app/index.js
[0] 
[0] ERROR in ./app/components/Onboarding/Autopilot.js
[0] Module not found: Error: Can't resolve 'react-icons/lib/fa' in '/home/pi/source/zap-desktop/app/components/Onboarding'
[0]  @ ./app/components/Onboarding/Autopilot.js 15:10-39
[0]  @ ./app/components/Onboarding/Onboarding.js
[0]  @ ./app/components/Onboarding/index.js
[0]  @ ./app/containers/Root.js
[0]  @ ./app/index.js
[0]  @ multi (webpack)-dev-server/client?http://localhost:1212 webpack/hot/dev-server react-hot-loader/patch webpack-dev-server/client?http://localhost:1212/ webpack/hot/only-dev-server ./app/index.js
[0] 
[0] ERROR in ./app/components/Onboarding/ConnectionType.js
[0] Module not found: Error: Can't resolve 'react-icons/lib/fa' in '/home/pi/source/zap-desktop/app/components/Onboarding'
[0]  @ ./app/components/Onboarding/ConnectionType.js 15:10-39
[0]  @ ./app/components/Onboarding/Onboarding.js
[0]  @ ./app/components/Onboarding/index.js
[0]  @ ./app/containers/Root.js
[0]  @ ./app/index.js
[0]  @ multi (webpack)-dev-server/client?http://localhost:1212 webpack/hot/dev-server react-hot-loader/patch webpack-dev-server/client?http://localhost:1212/ webpack/hot/only-dev-server ./app/index.js
[0] 
[0] ERROR in ./app/components/Wallet/Wallet.js
[0] Module not found: Error: Can't resolve 'react-icons/lib/fa' in '/home/pi/source/zap-desktop/app/components/Wallet'
[0]  @ ./app/components/Wallet/Wallet.js 15:10-39
[0]  @ ./app/components/Wallet/index.js
[0]  @ ./app/routes/activity/components/Activity.js
[0]  @ ./app/routes/activity/containers/ActivityContainer.js
[0]  @ ./app/routes/activity/index.js
[0]  @ ./app/routes.js
[0]  @ ./app/containers/Root.js
[0]  @ ./app/index.js
[0]  @ multi (webpack)-dev-server/client?http://localhost:1212 webpack/hot/dev-server react-hot-loader/patch webpack-dev-server/client?http://localhost:1212/ webpack/hot/only-dev-server ./app/index.js
[0] 
[0] ERROR in ./app/components/Contacts/ContactModal.js
[0] Module not found: Error: Can't resolve 'react-icons/lib/md' in '/home/pi/source/zap-desktop/app/components/Contacts'
[0]  @ ./app/components/Contacts/ContactModal.js 25:10-39
[0]  @ ./app/routes/app/components/App.js
[0]  @ ./app/routes/app/containers/AppContainer.js
[0]  @ ./app/routes/app/index.js
[0]  @ ./app/routes.js
[0]  @ ./app/containers/Root.js
[0]  @ ./app/index.js
[0]  @ multi (webpack)-dev-server/client?http://localhost:1212 webpack/hot/dev-server react-hot-loader/patch webpack-dev-server/client?http://localhost:1212/ webpack/hot/only-dev-server ./app/index.js
[0] 
[0] ERROR in ./app/components/GlobalError/GlobalError.js
[0] Module not found: Error: Can't resolve 'react-icons/lib/md' in '/home/pi/source/zap-desktop/app/components/GlobalError'
[0]  @ ./app/components/GlobalError/GlobalError.js 17:10-39
[0]  @ ./app/components/GlobalError/index.js
[0]  @ ./app/routes/app/components/App.js
[0]  @ ./app/routes/app/containers/AppContainer.js
[0]  @ ./app/routes/app/index.js
[0]  @ ./app/routes.js
[0]  @ ./app/containers/Root.js
[0]  @ ./app/index.js
[0]  @ multi (webpack)-dev-server/client?http://localhost:1212 webpack/hot/dev-server react-hot-loader/patch webpack-dev-server/client?http://localhost:1212/ webpack/hot/only-dev-server ./app/index.js
[0] 
[0] ERROR in ./app/components/ModalRoot/ModalRoot.js
[0] Module not found: Error: Can't resolve 'react-icons/lib/md' in '/home/pi/source/zap-desktop/app/components/ModalRoot'
[0]  @ ./app/components/ModalRoot/ModalRoot.js 17:10-39
[0]  @ ./app/components/ModalRoot/index.js
[0]  @ ./app/routes/app/components/App.js
[0]  @ ./app/routes/app/containers/AppContainer.js
[0]  @ ./app/routes/app/index.js
[0]  @ ./app/routes.js
[0]  @ ./app/containers/Root.js
[0]  @ ./app/index.js
[0]  @ multi (webpack)-dev-server/client?http://localhost:1212 webpack/hot/dev-server react-hot-loader/patch webpack-dev-server/client?http://localhost:1212/ webpack/hot/only-dev-server ./app/index.js

Ok.  They updated package 20.23.1 to fix the error.
I altered the zap package.json to have 20.23.1 for electron builder as it's dependency.

Now it gets the follow error:
Error: /usr/bin/node exited with code 1
Output:
yarn install v1.7.0
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
$ npm rebuild --runtime=electron --target=2.0.2 --disturl=https://atom.io/download/electron

> grpc@1.12.4 install /home/pi/source/zap-desktop/app/node_modules/grpc
> node-pre-gyp install --fallback-to-build --library=static_library

Failed to execute '/usr/bin/node /usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js clean' (1)
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.

Error output:
(node:16824) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
warning " > react-icons@2.2.5" has unmet peer dependency "react@^0.14.0 || ^15.0.0".
warning " > react-icons@2.2.5" has unmet peer dependency "react-dom@^0.14.0 || ^15.0.0".
warning "react-icons > react-icon-base@2.0.7" has unmet peer dependency "react@*".
warning "react-icons > react-icon-base@2.0.7" has unmet peer dependency "react-dom@*".
warning "react-icons > react-icon-base@2.0.7" has unmet peer dependency "prop-types@*".
node-pre-gyp WARN Tried to download(403): https://storage.googleapis.com/grpc-precompiled-binaries/node/grpc/v1.12.4/electron-v2.0-linux-arm-glibc.tar.gz 
node-pre-gyp WARN Pre-built binaries not found for grpc@1.12.4 and electron@2.0.2 (electron-v2.0 ABI, glibc) (falling back to source compile with node-gyp) 
gyp ERR! clean error 
gyp ERR! stack Error: EACCES: permission denied, rmdir 'build'
gyp ERR! System Linux 4.14.30-v7+
gyp ERR! command "/usr/bin/node" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "clean"
gyp ERR! cwd /home/pi/source/zap-desktop/app/node_modules/grpc
gyp ERR! node -v v10.4.1
gyp ERR! node-gyp -v v3.6.2
gyp ERR! not ok 
node-pre-gyp ERR! build error 
node-pre-gyp ERR! stack Error: Failed to execute '/usr/bin/node /usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js clean' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/home/pi/source/zap-desktop/app/node_modules/grpc/node_modules/node-pre-gyp/lib/util/compile.js:83:29)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:961:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:248:5)
node-pre-gyp ERR! System Linux 4.14.30-v7+
node-pre-gyp ERR! command "/usr/bin/node" "/home/pi/source/zap-desktop/app/node_modules/grpc/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build" "--library=static_library"
node-pre-gyp ERR! cwd /home/pi/source/zap-desktop/app/node_modules/grpc
node-pre-gyp ERR! node -v v10.4.1
node-pre-gyp ERR! node-pre-gyp -v v0.10.0
node-pre-gyp ERR! not ok 
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! grpc@1.12.4 install: `node-pre-gyp install --fallback-to-build --library=static_library`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the grpc@1.12.4 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2018-07-18T09_12_20_139Z-debug.log
error Command failed with exit code 1.

    at ChildProcess.childProcess.once.code (/home/pi/source/zap-desktop/node_modules/builder-util/src/util.ts:251:14)
    at Object.onceWrapper (events.js:273:13)
    at ChildProcess.emit (events.js:182:13)
    at maybeClose (internal/child_process.js:961:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:248:5)
From previous event:
    at installOrRebuild (/home/pi/source/zap-desktop/node_modules/electron-builder-lib/out/util/yarn.js:68:17)
    at /home/pi/source/zap-desktop/node_modules/electron-builder/src/cli/install-app-deps.ts:56:9
    at Generator.next (<anonymous>)
    at runCallback (timers.js:696:18)
    at tryOnImmediate (timers.js:667:5)
    at processImmediate (timers.js:649:5)
From previous event:
    at installAppDeps (/home/pi/source/zap-desktop/node_modules/electron-builder/out/cli/install-app-deps.js:174:17)
    at then (/home/pi/source/zap-desktop/node_modules/electron-builder/src/cli/cli.ts:42:48)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.

And this... is why I *HATE* linux.  I'm logged in as root... and ran the program with sudo... how much more do I need for Linux to actually BELIEVE me?

I'm now running a terminal with sudo -i.

I deleted the entire zap-desktop directory, and starting over.
git clone https://github.com/LN-Zap/zap-desktop.git
CD zap-desktop
<I edited the package.json to give electron builder 20.23.1 as it's dependency>
yarn

Result:
Error: /usr/bin/node exited with code 1
Output:
yarn install v1.7.0
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
$ npm rebuild --runtime=electron --target=2.0.2 --disturl=https://atom.io/download/electron

> grpc@1.12.4 install /home/pi/source/zap-desktop/app/node_modules/grpc
> node-pre-gyp install --fallback-to-build --library=static_library

Failed to execute '/usr/bin/node /usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js clean' (1)
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.

Error output:
(node:32200) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
warning " > react-icons@2.2.5" has unmet peer dependency "react@^0.14.0 || ^15.0.0".
warning " > react-icons@2.2.5" has unmet peer dependency "react-dom@^0.14.0 || ^15.0.0".
warning "react-icons > react-icon-base@2.0.7" has unmet peer dependency "react@*".
warning "react-icons > react-icon-base@2.0.7" has unmet peer dependency "react-dom@*".
warning "react-icons > react-icon-base@2.0.7" has unmet peer dependency "prop-types@*".
node-pre-gyp WARN Tried to download(403): https://storage.googleapis.com/grpc-precompiled-binaries/node/grpc/v1.12.4/electron-v2.0-linux-arm-glibc.tar.gz 
node-pre-gyp WARN Pre-built binaries not found for grpc@1.12.4 and electron@2.0.2 (electron-v2.0 ABI, glibc) (falling back to source compile with node-gyp) 
gyp ERR! clean error 
gyp ERR! stack Error: EACCES: permission denied, rmdir 'build'
gyp ERR! System Linux 4.14.30-v7+
gyp ERR! command "/usr/bin/node" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "clean"
gyp ERR! cwd /home/pi/source/zap-desktop/app/node_modules/grpc
gyp ERR! node -v v10.4.1
gyp ERR! node-gyp -v v3.6.2
gyp ERR! not ok 
node-pre-gyp ERR! build error 
node-pre-gyp ERR! stack Error: Failed to execute '/usr/bin/node /usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js clean' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/home/pi/source/zap-desktop/app/node_modules/grpc/node_modules/node-pre-gyp/lib/util/compile.js:83:29)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:961:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:248:5)
node-pre-gyp ERR! System Linux 4.14.30-v7+
node-pre-gyp ERR! command "/usr/bin/node" "/home/pi/source/zap-desktop/app/node_modules/grpc/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build" "--library=static_library"
node-pre-gyp ERR! cwd /home/pi/source/zap-desktop/app/node_modules/grpc
node-pre-gyp ERR! node -v v10.4.1
node-pre-gyp ERR! node-pre-gyp -v v0.10.0
node-pre-gyp ERR! not ok 
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! grpc@1.12.4 install: `node-pre-gyp install --fallback-to-build --library=static_library`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the grpc@1.12.4 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2018-07-18T10_17_42_264Z-debug.log
error Command failed with exit code 1.

    at ChildProcess.childProcess.once.code (/home/pi/source/zap-desktop/node_modules/builder-util/src/util.ts:251:14)
    at Object.onceWrapper (events.js:273:13)
    at ChildProcess.emit (events.js:182:13)
    at maybeClose (internal/child_process.js:961:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:248:5)
From previous event:
    at installOrRebuild (/home/pi/source/zap-desktop/node_modules/electron-builder-lib/out/util/yarn.js:68:17)
    at /home/pi/source/zap-desktop/node_modules/electron-builder/src/cli/install-app-deps.ts:56:9
    at Generator.next (<anonymous>)
    at runCallback (timers.js:696:18)
    at tryOnImmediate (timers.js:667:5)
    at processImmediate (timers.js:649:5)
From previous event:
    at installAppDeps (/home/pi/source/zap-desktop/node_modules/electron-builder/out/cli/install-app-deps.js:174:17)
    at then (/home/pi/source/zap-desktop/node_modules/electron-builder/src/cli/cli.ts:42:48)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.

--- a/package.json
+++ b/package.json
@@ -171,7 +171,7 @@
     "cross-spawn": "^6.0.5",
     "csp-html-webpack-plugin": "^2.3.0",
     "css-loader": "^0.28.11",
-    "electron-builder": "^20.21.2",
+    "electron-builder": "^20.24.5",
     "electron-devtools-installer": "^2.2.4",
     "enzyme": "^2.9.1",
     "enzyme-to-json": "^1.5.1",

